### PR TITLE
GH-4907 Move Client.updatePost call into an action

### LINKED ISSUE #4907: [Help Wanted] Move instances of Client.updatePost() in components to an action
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

---

**Difficulty**: Introductory (new to codebase)

**Details**:
To be [Flux](https://facebook.github.io/flux/docs/actions-and-the-dispatcher.html)-compliant our web utilities should be separate from our views and UI components. Views should use re-usable actions instead of directly interacting with the web utilities (specifically Client/client.jsx in our case).

Move any instances of `Client.updatePost()` inside of the [components directory](https://github.com/mattermost/platform/tree/master/webapp/components) into an action inside of [webapp/actions/post_actions.jsx](https://github.com/mattermost/platform/blob/master/webapp/actions/post_actions.jsx) and use the new action in place of where the `Client.updatePost()` invocations were.

Any common functionality should be performed in the action where possible, for example event dispatches through `AppDispatcher`.

Please see other actions inside of [webapp/actions/post_actions.jsx](https://github.com/mattermost/platform/blob/master/webapp/actions/post_actions.jsx) for examples.

A quick grep at the time of creation of this ticket shows `Client.updatePost()` being used in:
- https://github.com/mattermost/platform/tree/master/webapp/components/edit_post_modal.jsx

### PR DESCRIPTION
**Pull request by mikaoelitiana**
#### Summary
Client.updatePost where called from components so they needed to be moved inside action and call that action from component instead

#### Ticket Link
fixes #4907

#### Checklist
- [x] Has UI changes

### CODE REVIEW

File: webapp/components/edit_post_modal.jsx [hunk 1]
```diff
@@ -91,16 +89,12 @@ export default class EditPostModal extends React.Component {
             return;
         }
 
-        Client.updatePost(
+        updatePost(
             updatedPost,
             () => {
                 loadPosts(updatedPost.channel_id);
```
Comment (jwilander) [inline webapp/components/edit_post_modal.jsx]: Could you move the `loadPosts` line into the success callback of the action itself?

Comment (mikaoelitiana) [inline webapp/components/edit_post_modal.jsx]: I will change that quickly

File: webapp/actions/post_actions.jsx [hunk 2]
```diff
@@ -363,6 +363,15 @@ export function createPost(post, doLoadPost, success, error) {
     );
 }
 
+export function updatePost(post, success) {
+    Client.updatePost(
+        post,
+        success,
```
Comment (jwilander) [inline webapp/actions/post_actions.jsx]: I wouldn't worry about squashing the commits, but can you add the `loadPosts` call in here? It should look similar to this https://github.com/mattermost/platform/blob/master/webapp/actions/channel_actions.jsx#L194

**Comment by mikaoelitiana** [comment]
@jwilander I made the required changes

**Comment by mikaoelitiana** [comment]
Thanks!! Loved working on it

**Pull request merged**